tools/clang/unittests/HLSLExec/LongVectors.cpp

```diff
@@ -0,0 +1,316 @@
+#include "LongVectors.h"
+#include "HlslExecTestUtils.h"
+#include <iomanip>
+
+LongVector::BinaryOpType
+LongVector::getBinaryOpType(const std::wstring &OpTypeString) {
+  return getLongVectorOpType<LongVector::BinaryOpType>(
+      binaryOpTypeStringToEnumMap, OpTypeString,
+      std::size(binaryOpTypeStringToEnumMap));
+}
+
+LongVector::UnaryOpType
+LongVector::getUnaryOpType(const std::wstring &OpTypeString) {
+  return getLongVectorOpType<LongVector::UnaryOpType>(
+      unaryOpTypeStringToEnumMap, OpTypeString,
+      std::size(unaryOpTypeStringToEnumMap));
+}
+
+// These are helper arrays to be used with the TableParameterHandler that parses
+// the LongVectorOpTable.xml file for us.
+static TableParameter BinaryOpParameters[] = {
+    {L"DataType", TableParameter::STRING, true},
+    {L"OpTypeEnum", TableParameter::STRING, true},
+    {L"InputValueSetName1", TableParameter::STRING, false},
+    {L"InputValueSetName2", TableParameter::STRING, false},
+};
+
+static TableParameter UnaryOpParameters[] = {
+    {L"DataType", TableParameter::STRING, true},
+    {L"OpTypeEnum", TableParameter::STRING, true},
+    {L"InputValueSetName1", TableParameter::STRING, false},
+};
+
+bool LongVector::OpTest::classSetup() {
+  // Run this only once.
+  if (!Initialized) {
+    Initialized = true;
+
+    HMODULE Runtime = LoadLibraryW(L"d3d12.dll");
```

Comment:
Where is this library freed?

Reply:
Good question, I wondered the same. This setup code is copied from
ExecutionTest.cpp and has the note below to not call FreeLibrary on the handle.
Which I now realize has a typo from the naming update. I'll do a little digging
and report back - this is worth understanding to make sure we aren't leaking the
library handle.

Reply:
https://github.com/microsoft/DirectXShaderCompiler/issues/7583

Reply:
Well, I was trying the 'github pull requests' plugin for vscode out. Looks like its reply to comments just submits new comments. Pasting them here.

Good question, I wondered the same. This setup code is copied from
ExecutionTest.cpp and has the note below to not call FreeLibrary on the handle.
Which I now realize has a typo from the naming update. I'll do a little digging
and report back - this is worth understanding to make sure we aren't leaking the
library handle.

https://github.com/microsoft/DirectXShaderCompiler/issues/7583